src/core/jit/error.rs

```diff
@@ -51,52 +56,67 @@ impl ErrorExtensions for Error {
             Error::ParseError(error) => error.extend(),
             Error::IR(error) => error.extend(),
             Error::Validation(error) => error.extend(),
-        }
-    }
-}
-
-impl Error {
-    pub fn path(&self) -> Vec<PathSegment> {
-        match self {
-            Error::Validation(error) => match error {
-                ValidationError::ScalarInvalid { type_of: _, path } => {
-                    vec![PathSegment::Field(path.clone())]
-                }
-            },
-            _ => Vec::new(),
+            Error::ServerError(error) => error.extend(),
         }
     }
 }
 
 pub type Result<A> = std::result::Result<A, Error>;
 
-impl Error {
-    fn into_server_error_with_pos(self, pos: Option<Pos>) -> ServerError {
-        let extensions = self.extend().extensions;
-        let mut server_error = ServerError::new(self.to_string(), pos);
+impl From<Error> for ServerError {
+    fn from(val: Error) -> Self {
+        // async_graphql::parser::Error has special conversion to ServerError
+        if let Error::ParseError(error) = val {
+            return error.into();
+        }
+
+        let extensions = val.extend().extensions;
+        let mut server_error = ServerError::new(val.to_string(), None);
 
         server_error.extensions = extensions;
-        server_error.path = self.path();
 
         server_error
     }
+}
 
-    pub fn into_server_error(self) -> ServerError {
-        match self {
-            // async_graphql::parser::Error has special conversion to ServerError
-            Error::ParseError(error) => error.into(),
-            error => error.into_server_error_with_pos(None),
+#[derive(Debug, Serialize, Clone)]
+pub struct LocationError<Err> {
+    pub error: Err,
+    pub pos: Pos,
+    pub path: Vec<PathSegment>,
+}
```

Comment:
```suggestion
pub struct Positioned<Value> {
    pub value: Value,
    pub pos: Pos,
    pub path: Vec<String>,
}
```

Can generalize it.

Reply:
done, moved to model and added `PathSegment` and  `Post`.